lib/omniauth/authenticity_token_protection.rb

```diff
@@ -1,4 +1,9 @@
-require 'rack-protection'
+begin
+  require 'rack-protection'
+rescue LoadError
+  # support rack/protection 3.x
+  require 'rack_protection'
+end
```

Comment:
Have you tried just a single require `require 'rack/protection`? Might work for both versions

Reply:
And wouldn't requiring `rack_protection` try to load the wrong gem? (If https://rubygems.org/gems/rack_protection would exist (hopefully it doesn't exist because the name is protected from squatting))

Reply:
Yes, that should work. Thanks for the suggestion.

Reply:
> And wouldn't requiring `rack_protection` try to load the wrong gem? (If https://rubygems.org/gems/rack_protection would exist (hopefully it doesn't exist because the name is protected from squatting))

It's worse than that: any gem could ship a file `rack_protection.rb` or `rack-protection.rb`.